File: 334-IncreasingTripletSubsequence/334-IncreasingTripletSubsequence.js
// Last updated: 26/12/2025, 16:01:06
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    let min = Infinity, max = Infinity;

    for(let n of nums){
        if(n <= min){
            min = n
        }else if(n <= max){
            max = n
        }else {
            return true
        }
    }
    return false
};

File: 334-IncreasingTripletSubsequence/334-IncreasingTripletSubsequence.java
// Last updated: 26/12/2025, 16:01:09
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int min = Integer.MAX_VALUE, max = Integer.MAX_VALUE;
        
        for(int num : nums){
            if(num <= min){
                min = num;
            }else if(num <= max){
                max = num;
            }else{
                return true;
            }
        }
        
        return false;
    }
}